### ISSUE TITLE
Unable to re-use nix-based build

### ISSUE THREAD

**Issue by blackheaven** [haskell-cryptography/libsodium-bindings#131]
I'm trying to add [nix build support on a project which depends on `sel`](https://github.com/haskell-cryptography/one-time-password/pull/6), but I fail to make *GHC* detect `sel`:

```
@nix { "action": "setPhase", "phase": "setupCompilerEnvironmentPhase" }
setupCompilerEnvironmentPhase
Build with /nix/store/psds2pz1qhlr4z8qcahqii6kq1xsawb8-ghc-9.4.8.
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/sbin4qzj03wn252n0qibwn6jj9p1xnrc-bvh2s0sx28rfp0gzysj4p5fwzxzz0djx-source
source root is bvh2s0sx28rfp0gzysj4p5fwzxzz0djx-source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "compileBuildDriverPhase" }
compileBuildDriverPhase
setupCompileFlags: -package-db=/build/tmp.2qEJsW8rMl/setup-package.conf.d -j16 +RTS -A64M -RTS -threaded -rtsopts
[1 of 2] Compiling Main             ( /nix/store/4mdp8nhyfddh7bllbi7xszz7k9955n79-Setup.hs, /build/tmp.2qEJsW8rMl/Main.o )
[2 of 2] Linking Setup
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
configureFlags: --verbose --prefix=/nix/store/bypr7jw18cg0la1mq0wb3wnhiq8rb1sq-one-time-password-3.0.0.0 --libdir=$prefix/lib/$compiler --libsubdir=$abi/$libname --docdir=/nix/store/pq6lxf7hii5ghjgc9fayl5cpln5c6w1k-one-time-password-3.0.0.0-doc/share/doc/one-time-password-3.0.0.0 --with-gcc=gcc --package-db=/build/tmp.2qEJsW8rMl/package.conf.d --ghc-options=-j16 +RTS -A64M -RTS --disable-split-objs --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --ghc-option=-split-sections --ghc-options=-haddock --extra-lib-dirs=/nix/store/5i51nfixxx3p3gshkfsjj4bzp7wajwxz-ncurses-6.4/lib --extra-lib-dirs=/nix/store/9kd4bc8fpclpvf1vdwlbila71svyb6w1-libffi-3.4.4/lib --extra-lib-dirs=/nix/store/s3s7gv33p88kzbgki2bprg2a1nc7jnf8-gmp-with-cxx-6.3.0/lib --extra-lib-dirs=/nix/store/84qfr9nk7c7yh1ah3p9nvlq9d8y3h0rx-sel-0.0.1.0/lib --extra-lib-dirs=/nix/store/bvavjfn6m8x5y5i7a8qf85knknc4jrbp-base16-1.0/lib --extra-lib-dirs=/nix/store/gpmsy9p2fffmpbfc6x6xz12w9cn27ll3-primitive-0.8.0.0/lib --extra-lib-dirs=/nix/store/s6wr8nm2cxkz25lawf51aw8dv2kdq26l-text-short-0.1.5/lib --extra-lib-dirs=/nix/store/mqi0rc4qb5sjs78fbbgwqpcna1m71hwj-hashable-1.4.3.0/lib --extra-lib-dirs=/nix/store/5w8sb907h7rid9amykak5d8kjf3za4gw-libsodium-bindings-0.0.1.0/lib --extra-lib-dirs=/nix/store/zriy44k8iya3j45lxhfm8zf4s5q2s7zg-text-display-0.0.5.1/lib
Using Parsec parser
Configuring one-time-password-3.0.0.0...
CallStack (from HasCallStack):
  withMetadata, called at libraries/Cabal/Cabal/src/Distribution/Simple/Utils.hs:370:14 in Cabal-3.8.1.0:Distribution.Simple.Utils
Error: Setup: Encountered missing or private dependencies:
sel
```

PS: it seems that freezing GHC version causes the issue

**Comment by Kleidukos** [comment]
Sorry I have no idea how I could help. Can't nix get sources from github?

**Comment by Kleidukos** [comment]
Fixed by #132
